components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/WorkflowsApiServiceImpl.java

```diff
@@ -0,0 +1,171 @@
+import org.wso2.carbon.apimgt.core.api.APIMgtAdminService;
+import org.wso2.carbon.apimgt.core.api.WorkflowExecutor;
+import org.wso2.carbon.apimgt.core.api.WorkflowResponse;
+import org.wso2.carbon.apimgt.core.exception.APIManagementException;
+import org.wso2.carbon.apimgt.core.exception.APIMgtResourceNotFoundException;
+import org.wso2.carbon.apimgt.core.exception.ExceptionCodes;
+import org.wso2.carbon.apimgt.core.models.WorkflowStatus;
+import org.wso2.carbon.apimgt.core.util.APIMgtConstants;
+import org.wso2.carbon.apimgt.core.workflow.Workflow;
+import org.wso2.carbon.apimgt.core.workflow.WorkflowExecutorFactory;
+import org.wso2.carbon.apimgt.rest.api.admin.*;
+import org.wso2.carbon.apimgt.rest.api.admin.dto.*;
+import org.wso2.carbon.apimgt.rest.api.admin.mappings.WorkflowMappingUtil;
+import org.wso2.carbon.apimgt.rest.api.common.dto.ErrorDTO;
+import org.wso2.carbon.apimgt.rest.api.common.util.RestApiUtil;
+
+import org.wso2.carbon.apimgt.rest.api.admin.NotFoundException;
+
+import org.wso2.msf4j.Request;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.core.Response;
+
+public class WorkflowsApiServiceImpl extends WorkflowsApiService {
+
+    private static final Logger log = LoggerFactory.getLogger(WorkflowsApiServiceImpl.class);
+
+    @Override
+    public Response workflowsGet(String accept, String ifNoneMatch, String ifModifiedSince, String type,
+            Request request) throws NotFoundException {
+        try {
+            APIMgtAdminService apiMgtAdminService = RestApiUtil.getAPIMgtAdminService();
+            List<Workflow> workflowList;
+            if (type == null) {
+                workflowList = apiMgtAdminService.retrieveUncompletedWorkflows();
+            } else {
+                workflowList = apiMgtAdminService.retrieveUncompletedWorkflowsByType(type);
+            }
+
+            WorkflowListDTO workflowListDTO = WorkflowMappingUtil.toWorkflowListDTO(workflowList);
+            return Response.ok().entity(workflowListDTO).build();
+        } catch (APIManagementException e) {
+            String errorMessage = "Error occurred while retrieving workflows list";
+            org.wso2.carbon.apimgt.rest.api.common.dto.ErrorDTO errorDTO = RestApiUtil.getErrorDTO(e.getErrorHandler());
+            log.error(errorMessage, e);
+            return Response.status(e.getErrorHandler().getHttpStatusCode()).entity(errorDTO).build();
+        }
+
+    }
+
+    @Override
+    public Response workflowsWorkflowReferenceIdGet(String workflowReferenceId, Request request)
+            throws NotFoundException {
+        try {
+            APIMgtAdminService apiMgtAdminService = RestApiUtil.getAPIMgtAdminService();
+            Workflow workflow = apiMgtAdminService.retrieveWorkflow(workflowReferenceId);
+            WorkflowDTO workflowDTO = WorkflowMappingUtil.toWorkflowDTO(workflow);
+            return Response.ok().entity(workflowDTO).build();
```

Comment:
Need to handle the 404 scenario. The best option is to handle it at the DAO layer itself by throwing a DAO Exception appropriately.